platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConverterService.java

```diff
@@ -98,24 +96,6 @@ class ConverterService implements ConfigLifecycle {
         this.converters = new ConcurrentHashMap<>();
     }
 
-    @NonNull
-    private <T, C extends ConfigConverter<T>> Class<T> getConverterType(@NonNull final Class<C> converterClass) {
```

Comment:
I would have expected that the given functionality will stay in place for now and the new one is added in addition.

Reply:
Yes. The impact would have been more contained if I added another method to the existing alternatives. Still, there would be a slight difference in the behavior of the two methods. While the new one accepted lambda objects (maybe not important for us), the existing one would fail on runtime for the same case, which seemed inconsistent (can be documented). Also, it seemed more straightforward to use if you don't have to reflect on which signature is better, and although less flexible, by only providing the option of having the types explicitly declared in the signature seemed less prone to errors.
 Happy to add the other signatures back.

Reply:
As discussed with @hendrikebbers , I will recover the removed methods add the `@deprecated` annotation and create a second ticket for its future removal and documentation changes. The reasoning behind this:  we want to make sure to have a workflow compatible with cases where we have external clients for this functionality, and an all-or-nothing approach is not suitable for the future